### PR TITLE
427 hardcoded font in design system is hard to override

### PR DESCRIPTION
**Pull request by keesvanopdorp**
Inplemented the necessary changes

### CODE REVIEW

File: src/components/nimbleAutoComplete/NimbleAutoComplete.tsx [hunk 1]
```diff
@@ -42,7 +42,7 @@ export const NimbleAutoComplete: React.FC<NimbleAutoCompleteProps> = ({
   labelSize = 14,
   labelWeight = '600',
   placeholder,
-  fontFamily = 'Roboto,Helvetica,Arial,sans-serif',
```
Comment (Manoj201) [inline src/components/nimbleAutoComplete/NimbleAutoComplete.tsx:45]: fontFamily is optional prop, so don't remove this default value if prop is optional

File: src/components/nimbleAutoComplete/NimbleAutoComplete.tsx [hunk 2]
```diff
@@ -42,7 +42,7 @@ export const NimbleAutoComplete: React.FC<NimbleAutoCompleteProps> = ({
...
+  fontFamily,
```
Comment (Manoj201) [inline src/components/nimbleAutoComplete/NimbleAutoComplete.tsx:45]: fontFamily is optional prop, so don't remove this default value if prop is optional

File: src/components/nimbleBasicDataTable/NimbleBasicDataTable.tsx [hunk 3]
```diff
@@ -39,7 +39,7 @@ export interface ColumnDataBasicTableType {
 export const NimbleBasicDataTable = ({
   columns,
   rows,
-  fontFamily = 'Arial',
+  fontFamily,
```
Comment (Manoj201) [inline src/components/nimbleBasicDataTable/NimbleBasicDataTable.tsx:42]: please add default value for optional prop

File: src/components/nimbleButton/NimbleButton.tsx [hunk 4]
```diff
@@ -131,7 +131,7 @@ export const NimbleButton: React.FC<NimbleButtonProps> = ({
   loading,
   onClick,
   icon,
-  fontFamily = 'Roboto,Helvetica,Arial,sans-serif',
+  fontFamily,
```
Comment (Manoj201) [inline src/components/nimbleButton/NimbleButton.tsx:134]: as above comment please fix this

File: src/components/nimbleChip/NimbleChip.tsx [hunk 5]
```diff
@@ -44,7 +44,7 @@ export const NimbleChip: React.FC<NimbleChipProps> = ({
   labelcolor = '#FFFFFE',
   fontWeight = '400',
   fontSize = '14px',
-  fontFamily = 'Arial',
+  fontFamily,
```
Comment (Manoj201) [inline src/components/nimbleChip/NimbleChip.tsx:47]: as above comment please fix this

File: src/components/nimbleDataGrid/NimbleDataGrid.tsx [hunk 6]
```diff
@@ -64,7 +64,7 @@ export const NimbleDataGrid: React.FC<NimbleDataGridProps> = ({
   height = '100%',
   header,
   headerFontWeight = '600',
-  fontFamily = 'Roboto,Helvetica,Arial,sans-serif',
+  fontFamily,
```
Comment (Manoj201) [inline src/components/nimbleDataGrid/NimbleDataGrid.tsx:67]: as above comment please fix this

File: src/components/nimbleDataTable/NimbleDataTable.tsx [hunk 7]
```diff
@@ -124,7 +124,7 @@ export const NimbleDataTable: React.FC<NimbleDataTableProps> = ({
   InputFieldBorderColor = '#9A9FA5',
   InputFieldActiveBoxShadow = '0px 0px 0px 2px #DBF2FB, 0px 0px 0px 1px #77CBED inset',
   InputFieldHoverBoxShadow = '0px 0px 0px 2px #dae3f0, 0px 0px 0px 1px #50606B inset',
-  fontFamily = 'Roboto,Helvetica,Arial,sans-serif',
+  fontFamily,
```
Comment (Manoj201) [inline src/components/nimbleDataTable/NimbleDataTable.tsx:127]: as above comment please fix this

File: src/components/nimbleFileUploader/NimbleFileUploader.tsx [hunk 8]
```diff
@@ -66,8 +66,8 @@ export const NimbleFileUploader: React.FC<NimbleFileUploaderProps> = ({
   onFileSelect,
   headerFontColor = '#A4C2DB',
   uploadedFiles,
-  fontFamily = 'Roboto,Helvetica,Arial,sans-serif',
-  uploadButtonFontFamily = 'Roboto,Helvetica,Arial,sans-serif',
+  fontFamily,
+  uploadButtonFontFamily,
```
Comment (Manoj201) [inline src/components/nimbleFileUploader/NimbleFileUploader.tsx:70]: as above comment please fix this

**Review by Manoj201 — CHANGES_REQUESTED** [review]
Every component font family is optional, so better to have default font family in the component level.

**Comment by keesvanopdorp** [comment]
> if you are not of the same opinion as me and Max we need to discuss this in a meeting to find  a better solution for this problem because I'm of the opinion that this is not behavior you want in a component library. Furthermore in now need with every instance of a component you need to specify the font family if you don't use roboto

**Comment by keesvanopdorp** [comment]
> if you are not of the same opinion as me and Max we need to discuss this in a meeting to find  a better solution for this problem for this because I'm still of the opinion that this is not behavior you want in a component library. Furthermore now you need every instance of a component you need to specify the font family if you don't use roboto or need to wrap every nimble component in a styled function and overwrite the font family

**Pull request closed without merging**